src/checks/ending_blank_line.rs

```diff
@@ -0,0 +1,67 @@
+use crate::checks::Check;
+use crate::common::*;
+
+pub(crate) struct EndingBlankLineChecker {
+    name: &'static str,
+    template: &'static str,
+}
+
+impl Default for EndingBlankLineChecker {
+    fn default() -> Self {
+        Self {
+            name: "EndingBlankLine",
+            template: "No blank line at the end of file",
+        }
+    }
+}
+
+impl EndingBlankLineChecker {
+    fn message(&self) -> String {
+        format!("{}: {}", self.name, self.template)
+    }
+}
+
+impl Check for EndingBlankLineChecker {
+    fn run(&mut self, line: &LineEntry) -> Option<Warning> {
+        if line.is_empty() {
```

Comment:
I would like to check here if the string has ending blank line. For example:
```rust
if line.raw_string.chars().any(|c| c == '\n') {
    return None;
}
```

Reply:
First, if we go back to `for (index, line) in reader.lines().enumerate()` then we will not have `\n` in` raw_string`.

Second, let me to explain my solution: 

In `checks` we work with the representation of files as `EntryLine` collections. 
And I think that this is a good model, a good abstraction. I like it.
I suppose that in this model the file containing `\n` is two entry lines, not one (and IDE shows this file as 2 lines).
Thus, I believe that the implementation of this check is more in the correct presentation of the file as a `EntryLine` collection.

I hope that I was able to clearly explain.

Reply:
I have made the check more clear.